common/src/app/common/types/file.cljc

```diff
@@ -162,21 +169,23 @@
       (dm/get-in component [:objects shape-id]))))
 
 (defn get-ref-shape
-  "Retrieve the shape in the component that is referenced by the
-  instance shape."
+  "Retrieve the shape in the component that is referenced by the instance shape."
   [file-data component shape]
   (when (:shape-ref shape)
     (get-component-shape file-data component (:shape-ref shape))))
 
-(defn get-component-shapes
-  "Retrieve all shapes of the component"
-  [file-data component]
-  (let [components-v2 (dm/get-in file-data [:options :components-v2])]
-    (if (and components-v2
-             (not (:deleted component))) ;; the deleted components have its children in the :objects property
-      (let [instance-page (get-component-page file-data component)]
-        (cph/get-children-with-self (:objects instance-page) (:main-instance-id component)))
-      (vals (:objects component)))))
+(defn find-ref-shape
+  "Locate the near component in the local file or libraries, and retrieve the shape
+   referenced by the instance shape."
+  [file page libraries shape]
+  (let [root-shape     (ctn/get-component-shape (:objects page) shape)
+        component-file (if (= (:component-file root-shape) (:id file))
+                         file
+                         (get libraries (:component-file root-shape)))
+        component      (when component-file
```

Comment:
Why not use the new resolve-component ?

Reply:
It should be good. But we need component-file to get the shape, later. So i will need to calculate it, anyway. And if I have the file, then it's easier to directly call get-component.